linalg/sparsemat.cpp

```diff
@@ -772,12 +772,13 @@ void SparseMatrix::AddMultTranspose(const Vector &x, Vector &y,
 
    if (!Finalized())
    {
-      double *yp = y.GetData();
+      double *yp = y.HostReadWrite();
+      const double *xp = x.HostRead();
```

Comment:
These changes are not HIP-specific, have they been tested with CUDA and with a CPU device?

Reply:
Yes, this came up when I was trying out the branch in my application. I tried out the code with both CPU and CUDA backends.